lib/index.js

```diff
@@ -21,57 +21,84 @@ const schema = Joi.object().keys({
   extractKey: Joi.func()
 }).requiredKeys('type', 'event', 'address', 'extractKey');
 
-const pendingRequests = new Map();
+function setupPreResponseExt(server) {
+  server.ext('onPreResponse', (request, reply) => {
+    const rlRequest = pendingRequests.getAndRemove(request.id);
 
-exports.register = function (server, options, next) {
-  Joi.validate(options, schema, { abortEarly: false }, (err, processedOptions) => {
-    if (err) { return next(err); }
+    if (rlRequest && rlRequest.isConformant()){
+      const headers = rlRequest.getResponseHeaders();
 
-    const event = processedOptions.event;
-    const type = processedOptions.type;
-    const extractKey = processedOptions.extractKey;
-    const onError = processedOptions.onError;
-    const limitd = new LimitdClient(processedOptions.address);
+      Object.keys(headers).forEach(
+        key => request.response.header(key, headers[key]));
+    }
 
-    limitd.on('error', () => {
-      // ignore. what to do on error is decided by on error
-    });
+    reply.continue();
+  });
+}
+
+function setupRateLimitEventExt(server, options) {
+  const event = options.event;
+  const type = options.type;
+  const extractKey = options.extractKey;
+  const onError = options.onError;
+
+  server.ext(event, (request, reply) => {
+    const limitd = limitdPool.get(options.address);
 
-    server.ext(event, (request, reply) => {
-      extractKey(request, reply, (err, key) =>{
-        if (err) { return reply(err); }
+    // If current response is already not conformant we can stop
+    const currentRlRequest = pendingRequests.get(request.id);
+    if (currentRlRequest && !currentRlRequest.isConformant()) { return; }
```

Comment:
nothing special, just preference for getter methods, will make it an attr if you prefer it